cypress/integration/plugins/dashboards-assistant/mds_assistant_feature_flag_spec.js

```diff
@@ -0,0 +1,352 @@
+
+        // Dashboard admin settings saved object should be visible
+        cy.getElementByTestId('savedObjectsTableRow row-_dashboard_admin')
+          .contains('_dashboard_admin')
+          .getElementByTestId('checkboxSelectRow-_dashboard_admin')
+          .click();
+
+        // Delete the dashboard admin settings saved object
+        cy.getElementByTestId('savedObjectsManagementDelete')
+          .should('exist')
+          .click();
+        cy.getElementByTestId('confirmModalConfirmButton')
+          .should('exist')
+          .click();
+
+        cy.visit(`${BASE_PATH}/app/settings`);
+
+        // AI feature toggle should be restored back to default
+        cy.getElementByTestId(
+          'advancedSetting-editField-enableAIFeatures'
+        ).should('have.attr', 'aria-checked', 'true');
+      });
+    });
+
+    describe('Non-dashboard admin user', () => {
+      const NONE_DASHBOARDS_ADMIN_USERNAME = 'workspace-test';
+      const WORKSPACE_TEST_ROLE_NAME = 'workspace-test-role';
+      const originalUser = ADMIN_AUTH.username;
+      const originalPassword = ADMIN_AUTH.password;
+
+      before(() => {
+        // Prepare non OSD admin user
+        cy.createInternalUser(
+          NONE_DASHBOARDS_ADMIN_USERNAME,
+          workspaceTestUser
+        );
+        cy.createRole(WORKSPACE_TEST_ROLE_NAME, workspaceTestRole);
+        cy.createRoleMapping(WORKSPACE_TEST_ROLE_NAME, {
+          users: [NONE_DASHBOARDS_ADMIN_USERNAME],
+        });
+      });
+
+      beforeEach(() => {
+        // Login as non OSD admin user
+        ADMIN_AUTH.newUser = NONE_DASHBOARDS_ADMIN_USERNAME;
+        ADMIN_AUTH.newPassword = workspaceTestUser.password;
+      });
+
+      after(() => {
+        ADMIN_AUTH.newUser = originalUser;
+        ADMIN_AUTH.newPassword = originalPassword;
+        cy.deleteRoleMapping(WORKSPACE_TEST_ROLE_NAME);
+        cy.deleteInternalUser(NONE_DASHBOARDS_ADMIN_USERNAME);
+        cy.deleteRole(WORKSPACE_TEST_ROLE_NAME);
+      });
+
+      it('cannot toggle AI feature flag', () => {
+        cy.visit(`${BASE_PATH}/app/settings`);
+
+        // The AI feature toggle should be unavailble to be toggled
+        cy.contains('Enable AI features');
+        cy.getElementByTestId(
+          'advancedSetting-editField-enableAIFeatures'
+        ).should('be.disabled');
+        cy.contains('This setting is controlled by dashboard admin only.');
+      });
+
+      it('cannot see admin settings saved object in assets page', () => {
+        cy.visit(`${BASE_PATH}/app/objects`);
+
+        // Dashboard admin settings saved object should not be visible
+        cy.getElementByTestId('savedObjectsTable').within(() => {
+          cy.get('.euiTableRow').should('be.visible');
+          cy.contains('_dashboard_admin').should('not.exist');
+        });
+      });
+    });
+  });
+}
+
+if (
+  Cypress.env('DASHBOARDS_ASSISTANT_ENABLED') &&
+  Cypress.env('CYPRESS_DATASOURCE_MANAGEMENT_ENABLED')
```

Comment:
I guess data source flag is not required for dashboard admin setting.

Reply:
Done, moved the dashboard admin setting tests out of this if block.